src/rashdf/plan.py

```diff
@@ -1117,6 +1121,106 @@ def reference_lines_timeseries_output(self) -> xr.Dataset:
         """
         return self.reference_timeseries_output(reftype="lines")
 
+    def observed_timeseries_input(self, vartype: str = "Flow") -> xr.Dataset:
+        """Return observed timeseries input data for reference lines and points from a HEC-RAS HDF plan file.
+
+        Parameters
+        ----------
+        vartype : str, optional
+            The type of observed data to retrieve. Must be either "Flow" or "Stage".
+            (default: "Flow")
+
+        Returns
+        -------
+        Dict[Site: str, xr.DataArray]
+            An xarray Dataset with observed timeseries data for both reference lines and reference points.
+
+            Coordinates:
+            - 'time': datetime64[ns]
+
+            Data variables:
+            - 'Flow' or 'Stage': float64
+        """
+
+        # Decode the contents of the DataFrame from utf-8
+        def decode_bytes(val):
```

Comment:
Instead of this function let's use `utils.convert_ras_hdf_value`. 
```python
>>> obs = plan_hdf[RasPlanHdf.OBS_FLOW_OUTPUT_PATH]["Ref Line: Denton-Justin_RL"]
>>> pd.DataFrame(obs[:]).map(convert_ras_hdf_value)
                    Date  Simulation Time  Value
0    2015-10-22 23:45:00        -0.010417   1.96
1    2015-10-23 00:00:00         0.000000   1.96
2    2015-10-23 00:15:00         0.010417   1.80
3    2015-10-23 00:30:00         0.020833   2.13
4    2015-10-23 00:45:00         0.031250   2.13
...                  ...              ...    ...
1926 2015-11-11 23:15:00        19.968750  10.50
1927 2015-11-11 23:30:00        19.979167  10.50
1928 2015-11-11 23:45:00        19.989583  10.50
1929 2015-11-12 00:00:00        20.000000  10.50
1930 2015-11-12 00:15:00        20.010417  10.50

[1931 rows x 3 columns]
```

Reply:
Done deal